### DISCUSSION TITLE
Can I avoid this `.route()` repetition handling both GET and POST with the same handler

### DISCUSSION THREAD

**Discussion by frederikhors** [tokio-rs/axum#1304] [Q&A] [answered]
_First of all let me thank you all for this awesome project! I'm almost crying with joy at how good it is to use it!_

### Now the question (_I'm still studying Rust and Axum... sorry_):

I'm using this code and I want to handle both `GET` and `POST` requests with the same handler. Is it possible to avoid that repetition?

```rust
let mut app = Router::new()
  .route("/", get(handler))

  // Can I avoid this `.route()` repetition handling both GET and POST with the same handler
  .route(
      format!("{}/*key", &reverse_proxy_url).as_str(),
      get(reverse_proxy_handler),
  )
  .route(
      format!("{}/*key", &reverse_proxy_url).as_str(),
      post(reverse_proxy_handler),
  );
```

### Second question:

Is there a better way than the below?

```rust
format!("{}/*key", &reverse_proxy_url).as_str(),
```

**Comment by jplatte** [comment, marked as answer]
Yes, you can do that using [routing::on](https://docs.rs/axum/latest/axum/routing/fn.on.html), or by using the chained HTTP method methods on `MethodRouter`:

```rust
use axum::{
    routing::on,
    Router,
    routing::MethodFilter,
};

// routing::on
let app = Router::new()
    .route("/", get(handler))
    .route(
        format!("{}/*key", &reverse_proxy_url).as_str(),
        on(MethodFilter::GET | MethodFilter::POST, handler),
    );

// MethodRouter::post
let app = Router::new()
    .route("/", get(handler))
    .route(
        format!("{}/*key", &reverse_proxy_url).as_str(),
        get(handler).post(handler),
    );
```

**Reply by frederikhors** [reply]
Thanks. Can you please answer the second question?

**Reply by davidpdrsn** [reply]
`.route(&format!("{}/*key", &reverse_proxy_url), ...)`

**Reply by frederikhors** [reply]
Great. What beautiful people you are. Helping others especially if novices and enthusiasts help the world! Thank you.
